lib/ansible/modules/cloud/vmware/vmware_host_firewall_manager.py

```diff
@@ -0,0 +1,219 @@
+
+        """
+        fw_change_list = []
+        results = dict(changed=False, rule_set_state=dict())
+        for host in self.hosts:
+            firewall_system = host.configManager.firewallSystem
+            if firewall_system:
+                results['rule_set_state'][host.name] = dict()
+                for rule_option in self.rule_options:
+                    rule_name = rule_option.get('name', None)
+                    rule_enabled = rule_option.get('enabled', None)
+                    if rule_name is not None and rule_name in self.firewall_facts[host.name].keys():
+                        current_rule_state = self.firewall_facts[host.name][rule_name]['enabled']
+                        if rule_enabled is not None:
+                            if current_rule_state != rule_enabled:
+                                try:
+                                    if rule_enabled:
+                                        firewall_system.EnableRuleset(id=rule_name)
+                                    else:
+                                        firewall_system.DisableRuleset(id=rule_name)
+                                    fw_change_list.append(True)
+                                except vim.fault.NotFound as not_found:
+                                    self.module.fail_json(msg="Failed to enable rule set %s as"
+                                                              " rule set id is unknown : %s" % (rule_name,
+                                                                                                to_native(not_found.msg)))
+                                except vim.fault.HostConfigFault as host_config_fault:
+                                    self.module.fail_json(msg="Failed to enabled rule set %s as an internal"
+                                                              "error happened while reconfiguring"
+                                                              " rule set : %s" % (rule_name,
+                                                                                  to_native(host_config_fault.msg)))
+                            results['rule_set_state'][host.name][rule_name] = dict(current_state=rule_enabled,
+                                                                                   previous_state=current_rule_state,
+                                                                                   desired_state=rule_enabled,
+                                                                                   )
+                        else:
+                            self.module.fail_json(msg="Please specify rules.enabled for rule set"
+                                                      " %s as it is required parameter." % rule_name)
+                    else:
+                        self.module.fail_json(msg="Please specify rule.name for rule set"
+                                                  " as it is required parameter.")
```

Comment:
instead of: 
```
for host in self.hosts:
    …
```
I would have write:
```
for host in self.hosts:
    firewall_system = host.configManager.firewallSystem
    if firewall_system is None:
    	continue

    results['rule_set_state'][host.name] = dict()

    for rule_option in self.rule_options:

        rule_name = rule_option.get('name', None)
	if rule_name is None:
            self.module.fail_json(msg="Please specify rule.name for rule set"
                                      " as it is required parameter.")
        if rule_name not in self.firewall_facts[host.name]:
            self.module.fail_json(msg="rule named '%s' wasn't found." % rule_name)

        rule_enabled = rule_option.get('enabled', None)
	if rule_enabled is None:
            self.module.fail_json(msg="Please specify rules.enabled for rule set"
                                      " %s as it is required parameter." % rule_name)

        current_rule_state = self.firewall_facts[host.name][rule_name]['enabled']
        if current_rule_state != rule_enabled:
            try:
                if rule_enabled:
                    firewall_system.EnableRuleset(id=rule_name)
                else:
                    firewall_system.DisableRuleset(id=rule_name)
                fw_change_list.append(True)
            except vim.fault.NotFound as not_found:
                self.module.fail_json(msg="Failed to enable rule set %s as"
                                          " rule set id is unknown : %s" % (rule_name,
                                                                            to_native(not_found.msg)))
            except vim.fault.HostConfigFault as host_config_fault:
                self.module.fail_json(msg="Failed to enabled rule set %s as an internal"
                                          "error happened while reconfiguring"
                                          " rule set : %s" % (rule_name,
                                                              to_native(host_config_fault.msg)))

        results['rule_set_state'][host.name][rule_name] = dict(current_state=rule_enabled,
                                                               previous_state=current_rule_state,
                                                               desired_state=rule_enabled,
                                                               )
```
@Akasurde what do you think ?

It's more readable and it returns an helpful error if the rule name is set but not found.

Reply:
agree.